doc/notes/mapping/9337_mapping.md

```diff
@@ -0,0 +1,258 @@
+alone, the PIM is kept free of technology choices which are specified
+during Design.
+
+In its simplest form a Design model can be a list of required
+functional and behavioral elements in the system.  At a richer
+level, a model can be formalized in an industry standardized
+modeling language such as UML and SysML [5].
+The most precise models are executable and can be shown to fulfill
+requirements by walk-through testing or even with automation.
+
+![UML and SysML](uml.gif)  
+
+## Design  
+Requirements specification focused on functional requirements; Design
+focuses on the technologies being used to satisfy the requirements.
+Design specifies _non-functional_ requirements.  _Non-functional_
+may seem to be an odd term; it means specifications that define the
+technology of the system rather than its required behavior.
+Examples of design elements include hardware technologies,
+microprocessors, network protocols, operating systems, and programming
+languages.
+
+Design supplies an architecture upon which the specified requirements
+can be fulfilled.
+
+As in Requirement Specification, Design also produces a list of elements.
+This list of elements represents the system architecture.
+This list is different from the list produced
+by Requirements Specification, but it is related.  Later in this paper
+we will explore the relationship between elements in Requirements and
+elements in Design.  We will map them together into the Implementation.
+Design is complete when each element listed in the Requirements
+Specification has an architectural element in the Design that can be
+used to implement it.
+
+#### Models  
+Again, MDA prescribes modeling.  A list of elements can be sufficient
+for simple systems.  But for complex systems, a more precise definition
+of the Design can be communicated in a model.
+A Platform Specific Model (PSM) is crafted that illustrates the
+elements in the system/software architecture.  Since Design focuses
+on the platform and technologies, the model is made specific to the
+platform.  This Design model is independent of functional requirements
+but is very specific to the target architecture chosen technologies.
+The model adheres to
+semantics that clearly define and constrain each element in the Design.
+Each element of the system/software architecture is illustrated in a
+model of the software architecture.
+
+With a model of Requirements and a model of Design, a model-to-model (M2M)
+transformation can be done that maps these model elements together
+into the Implementation.
+As a final step of the Design process a set of rules is defined.
+One rule is defined for each type of element in the Requirements.
+Each rule explains how to pair an element in the Requirements with
+an element in the Design.
+This set of mapping rules is the core of the Model Translation.
+These rules can be applied manually.
+
+With models, much of this mapping and translation can be automated.
```

Comment:
These two sentences could be combined. "These rules can be applied manually, but much of this mapping and translation of the models can be automated."